### DISCUSSION TITLE
🔐 Encrypted Notes

### DISCUSSION THREAD

**Discussion by ndavd** [zk-org/zk#151] [Ideas]
It would be nice to provide a similar [feature](https://github.com/xwmx/nb#password-protected-encrypted-notes-and-bookmarks).

**Comment by mickael-menu** [comment]
This would be a useful feature, but I don't see this being available soon.

As an alternative, if you're on macOS, you can use an [encrypted disk image](https://support.apple.com/guide/disk-utility/create-a-disk-image-dskutl11888/mac#dsku7bb3d28c) to host your notebook. This will encrypt the whole notebook and will work with any tools, including `zk`, as long as the disk image is mounted.

**Reply by ndavd** [reply]
Thanks for the reply.I'm on Linux though so I'm going to have to wait then xD

**Reply by manoelpqueiroz** [reply]
@mickael-menu, sorry for necrobumping this one, but I've recently started with Zk and I feel note encryption would be awesome. Since this original post, has your perception on this potential feature changed? What do you see as obstacles for implementing it in Zk?

**Comment by notusknot** [comment]
On linux using `age` and `tar`, I've created these simple functions to encrypt and decrypt directories:
```
edir() { tar -cz $1 | age -p > $1.tar.gz.age && rm -rf $1 &>/dev/null && echo "$1 encrypted" }
ddir() { age -d $1 | tar -xz && rm -rf $1 &>/dev/null && echo "$1 decrypted" }
```
You can drop them in your shell's config and run the `edir path/to/dir/` and `ddir path/to/dir/` commands to respectively encrypt and decrypt directories.
